beets/plugins.py

```diff
@@ -673,22 +675,35 @@ def get_artist(artists, id_key='id', name_key='name'):
             to concatenate for the artist string (containing all artists).
             Defaults to 'name'.
         :type name_key: str or int
+        :param join_key: Key or index corresponding to a field containing a
+            keyword to use for combining artists into a single string, for
+            example "Feat.", "Vs.", "And" or similar. The default is None
+            which keeps the default behaviour (comma-separated).
+        :type join_key: str or int
         :return: Normalized artist string.
         :rtype: str
         """
         artist_id = None
-        artist_names = []
-        for artist in artists:
+        artist_string = ""
+        # Count items (supports generators as well as lists, dicts, ...)
+        total = sum(1 for _ in artists)
```

Comment:
A different option for dealing with generators (iteratables, in general) passed here would be to eagerly convert the artists to a list. That is, you could do:

```py
artists_list = list(artists)
```

…and then simply use `len(artists_list)`.

Reply:
Will do that. Thanks!

I just realilsed that my way of solving it didn't work anyway! Sorry for not testing better. It seems a generator is exhausted when it was counted with the sum(...) hack (it's been looped through to the end). I can't loop through it then because it's empty! Alright, sorry again, will fix soon.

Reply:
Yep, that's exactly the case!